src/Microsoft.AspNet.Mvc.Razor/Compilation/CompilationOptionsProviderExtension.cs

```diff
@@ -32,5 +32,24 @@ public static class CompilationOptionsProviderExtension
                                                               applicationEnvironment.Configuration)
                                           .ToCompilationSettings(applicationEnvironment.RuntimeFramework);
         }
+
+        /// <summary>
+        /// Parses the <see cref="ICompilerOptions"/> for a project specified by <see cref="IProjectContext"/>
+        /// and returns a <see cref="CompilationSettings"/> used for Roslyn compilation.
+        /// </summary>
+        /// <param name="compilerOptionsProvider">
+        /// A <see cref="ICompilerOptionsProvider"/> that reads compiler options.
+        /// </param>
+        /// <param name="projectContext">The <see cref="IProjectContext"/> for the application being compiled.</param>
+        /// <returns>The <see cref="CompilationSettings"/> for the application being compiled.</returns>
+        public static CompilationSettings GetCompilationSettings(
+            [NotNull] this ICompilerOptionsProvider compilerOptionsProvider,
+            [NotNull] IProjectContext projectContext)
```

Comment:
The use case was for users to be use these extensions in their custom precompilation modules - it was primarily the reason we have the other extension method.